tests/test_job.py

```diff
@@ -436,3 +439,179 @@ def update_status(self) -> JobStatus:
+        ),
+        pytest.param(
+            Job(script_path=Path("test.py"), name="job_1"),
+            "1234567",
+            JobStatus.QUEUED,
+            "afterok:1234567",
+            nullcontext(),
+            id="single_job_instance",
+        ),
+        pytest.param(
+            [
+                Job(script_path=Path("horse_with.py"), name="job_1"),
+                Job(script_path=Path("no_name.py"), name="job_2"),
+            ],
+            ["1234567", "4567891"],
+            [JobStatus.QUEUED, JobStatus.QUEUED],
+            "afterok:1234567:4567891",
+            nullcontext(),
+            id="multiple_job_instance",
+        ),
+        pytest.param(
+            [
+                Job(script_path=Path("king_crimson.py"), name="job_1"),
+                Job(script_path=Path("crimson_king.py"), name="job_2"),
+            ],
+            ["1234567", "not_an_id"],
+            [JobStatus.QUEUED, JobStatus.QUEUED],
+            None,
+            pytest.raises(
+                ValueError,
+                match=r"Invalid job ID 'not_an_id'\. Job IDs must be 7 digits long\.",
+            ),
+            id="multiple_job_instance_invalid_one",
+        ),
+        pytest.param(
+            [
+                Job(script_path=Path("king_crimson.py"), name="job_1"),
+                "9876543",
+            ],
+            ["1234567", None],
+            [JobStatus.QUEUED, None],
+            "afterok:1234567:9876543",
+            nullcontext(),
+            id="job_and_string_input",
+        ),
+        pytest.param(
+            Job(script_path=Path("test.py"), name="tornero"),
+            "1234567",
+            JobStatus.UNPREPARED,
+            None,
+            pytest.raises(
+                ValueError,
+                match="Job 'tornero' has not been submitted yet.",
+            ),
+            id="unprepared_job_instance",
+        ),
+        pytest.param(
+            [
+                Job(script_path=Path("script.py"), name="dalida"),
+                Job(script_path=Path("script.py"), name="mourir_sur_scene"),
+            ],
+            ["1234567", "4567896"],
+            [JobStatus.QUEUED, JobStatus.PREPARED],
+            None,
+            pytest.raises(
+                ValueError,
+                match="Job 'mourir_sur_scene' has not been submitted yet.",
+            ),
+            id="multiple_job_instance_one_not_submitted",
+        ),
+    ],
+)
+
+def test_build_dependency_string_with_string_input(
+    dependency: str | list[str] | Job | list[Job],
+    job_id: str | list[str] | None,
+    job_status: JobStatus,
+    expected: str | None,
+    expected_exception: type[Exception],
+) -> None:
+    """Test building dependency string from string and Job inputs."""
+    deps = dependency if isinstance(dependency, list) else [dependency]
+    ids = job_id if isinstance(job_id, list) else [job_id]
+    status = job_status if isinstance(job_id, list) else [job_status]
```

Comment:
If you don't want to bother with it, you can already specify lists in your tests.